ca/server.go

```diff
@@ -32,6 +34,45 @@ type APISecurityConfigUpdater interface {
 
 var _ APISecurityConfigUpdater = &Server{}
 
+// ExternalCATLSUpdater is a way to notify the server of TLS updates for the external CA
+type ExternalCATLSUpdater interface {
+	NotifyTLSChange() <-chan events.Event
+	GetCertificates() []tls.Certificate
+}
+
+// SecurityConfigTLSUpdater takes a security config and sets a watch so we can tell when TLS credentials have changed
+type SecurityConfigTLSUpdater struct {
```

Comment:
I'd suggest removing the abstraction until we get to that point. Then we can see what makes most sense. Passing a channel may be best, especially if the lifecycle of the watch can be managed by something external to the CA server.